src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -82,27 +82,26 @@ internal class AssemblyLoader
             return discoveredAssemblyLocations;
         }
 #endif
-        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger) where T : class
+        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger, IServiceProvider serviceProvider) where T : class
         {
             try
             {
                 var assembly = Assembly.Load(new AssemblyName(assemblyName));
                 var foundType =
                     TypeUtils.GetTypes(
                         assembly,
-                        type =>
-                        typeof(T).IsAssignableFrom(type) && !type.GetTypeInfo().IsInterface
-                        && type.GetConstructor(Type.EmptyTypes) != null, logger).FirstOrDefault();
+                        type => typeof(T).IsAssignableFrom(type) && !type.GetTypeInfo().IsInterface,
+                        logger).FirstOrDefault();
                 if (foundType == null)
                 {
                     return null;
                 }
 
-                return (T)Activator.CreateInstance(foundType, true);
+                return (T)ActivatorUtilities.CreateInstance(serviceProvider, foundType);
```

Comment:
Breaking change. We need to update changelog for these changes on how we load stuff.